docs/source/conf.py

```diff
@@ -106,13 +106,28 @@
               ]
 
 
+def ultimateReplace(app, docname, source):
+  result = source[0]
+  for key in app.config.ultimate_replacements:
+    result = result.replace(key, app.config.ultimate_replacements[key])
+  source[0] = result
+
+
+ultimate_replacements = {
+    "|version|": version,
+}
+
+
```

Comment:
In https://dirac--5353.org.readthedocs.build/en/5353/UserGuide/GettingStarted/InstallingClient/index.html I see:

> ./dirac-install.py -r 7.2.23.dev1+ge5d3cb897.d20210820 -t client

can you explain the magic?

Reply:
The part `|version|` is replaced by the dirac version. In my local tests this worked nicely to v7r2p22, but of course readthedocs is now python3, so this needs some work to actually replace this by the python 2 version string.

Reply:
well my question is where version is calculated.

Reply:
I am just wondering the same, but this @chrisburr has to tell us :)

Reply:
side question for @chrisburr is the patch number on the branch always going to be +1 compared to the last released patch?

Reply:
If this is just for Python 2 I would use `git describe --abbrev=0 --tags --match 'v[0-9][0-9]r*'` to do it properly rather than constructing it from the Python 3 version.

Reply:
For Python 3 it follows [this logic to construct it](https://pypi.org/project/setuptools-scm/#default-versioning-scheme) based on what's allowed in [PEP-440](https://www.python.org/dev/peps/pep-0440/).